Convert `activitypub/*` controller specs to request specs

mjankowski opened the pull request:
Similar to https://github.com/mastodon/mastodon/pull/33975 in that I'm doing a related batch here all at once. Not sure about diff here ... reviewing one commit at a time may be easier?

Most of this barely needed changing to move, but some notes:

- Converted in a few spots to use the `sign_with` helper instead of stubbing controller method, and authorized fetch mode stubbing level. 
- Combined a few examples for some small factory reduction
- In the outboxes spec, some assertions changed from looking for 'max-age=60, private' to looking for 'private, no-store' ... I believe this is more accurate and that the previous spec was passing because at the controller-only level the method does set via `expires_in`, but that value is overwritten when running the full stack. I believe that was the only change of assertion when converting.
- Misc style/formatting cleanup to keep things with consistent with rest of suite while doing the move

mjankowski committed:
Move AP controller specs to requests

mjankowski committed:
Actually convert/adjust

mjankowski committed:
Move file names

mjankowski commented:
Let me know if this is easier reviewed in smaller pieces -- specifically, if the portions which needed the new helper method should be pulled out first?

ClearlyClaire commented:
Hi! Sorry for the late reply.

I think most of the changes are good but the inboxes controller is more involved and `RAW_POST_DATA` is rather awkward. Can you split those in a separate PR?

mjankowski commented:
Extracted https://github.com/mastodon/mastodon/pull/34292 and found a way to do that w/out the raw post data approach.

ClearlyClaire commented:
Hm, the changes here still include the inboxes specs change

mjankowski commented:
I was going to let the other review/merge first, then rebase this to be just the other stuff.

mjankowski commented:
Rebased after inbox one merged, this is now the rest of the ap/* ones. Also updated description.

The change of the cache control headers in outboxes spec, is probably the only interesting thing remaining here. The rest is pretty straightforward conversion/formatting stuff.
